client/src/pages/Talks.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect, useContext, useRef } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link } from '@reach/router';
+
+import TalkCard from '../components/TalkCard';
+import Title from '../components/Title';
+import Spinner from '../components/Spinner';
+import UserContext from '../components/UserContext';
+
+import fireModal from '../utils/fireModal';
+import config from '../utils/config';
+
+const Talks = (props) => {
+  const [ talks, setTalks ] = useState([]);
+  const [ loaded, setLoaded ] = useState(false);
+  const [ active, setActive ] = useState(null);
+
+  const shouldFocus = useRef(null);
+
+  const setVotes = (talkId, newVotes) => {
+    const talk = talks.find(talk => talk.id === talkId);
+    talk.votes = newVotes;
+    setTalks(talks);
+    setActive(null);
+    shouldFocus.current.focus();
+    setActive(talkId);
+  };
+
+  const { user } = useContext(UserContext);
+
+  useEffect(() => {
+    setLoaded(false);
+
+    fetch(`${config.apiHost}/talks`)
+    .then(async res => {
+      const json = await res.json();
+
+      const { talks } = json;
+
+      const newTalks = Object.keys(talks)
+        .map(k => {
+          const t = talks[k];
+          t.id = k;
+          return t;
+        });
+
+      setTalks(newTalks);
+
+      setLoaded(true);
+    });
```

Comment:
Yes, thanks for pointing that out! Definitely needs some error handling, otherwise it just keeps displaying the spinner. Will fix that.

Reply:
Yes, thanks for pointing that out! Definitely needs some error handling, otherwise it just keeps displaying the spinner on network error. Will fix that.